scrypto-unit/src/test_runner.rs

```diff
@@ -396,16 +384,29 @@ pub struct TestRunner<E: NativeVmExtension, D: TestDatabase> {
     next_transaction_nonce: u32,
     trace: bool,
     state_hash_support: Option<StateHashSupport>,
-    collected_events: Option<Vec<Vec<(EventTypeIdentifier, Vec<u8>)>>>,
+    collected_events: Vec<Vec<(EventTypeIdentifier, Vec<u8>)>>,
+    xrd_free_credits_used: bool,
 }
 
 #[cfg(feature = "post_run_db_check")]
 impl<E: NativeVmExtension, D: TestDatabase> Drop for TestRunner<E, D> {
     fn drop(&mut self) {
-        let results = self
-            .check_db::<ResourceChecker>()
+        let db_results = self
+            .check_db::<ResourceDatabaseChecker>()
             .expect("Database should be consistent after running test");
-        println!("{:?}", results);
+        println!("{:#?}", db_results);
+
+        let event_results = SystemEventChecker::<ResourceEventChecker>::new()
+            .check_all_events(&self.database, &self.collected_events)
+            .expect("Events should be consistent");
+        println!("{:#?}", event_results);
+
+        // If free credits (xrd from thin air) have been used then reconciliation will fail
+        // due to missing mint events
```

Comment:
Are we planning to expose `free_credit_used_in_xrd` to receipt, so we can enable recon for all? Happy either way :)

Reply:
Ran into some complications with `free_credit_used_in_xrd` so avoided this for now :)